codegen-renderers/src/main/kotlin/io/vrap/codegen/languages/java/commands/JavaCommandsRenderer.kt

```diff
@@ -0,0 +1,178 @@
+package io.vrap.codegen.languages.java.commands
+
+import io.vrap.codegen.languages.extensions.EObjectExtensions
+import io.vrap.codegen.languages.java.extensions.JavaObjectTypeExtensions
+import io.vrap.codegen.languages.java.extensions.resource
+import io.vrap.codegen.languages.java.extensions.returnType
+import io.vrap.codegen.languages.java.extensions.toRequestName
+import io.vrap.rmf.codegen.io.TemplateFile
+import io.vrap.rmf.codegen.rendring.MethodRenderer
+import io.vrap.rmf.codegen.rendring.utils.escapeAll
+import io.vrap.rmf.codegen.rendring.utils.keepIndentation
+import io.vrap.rmf.codegen.types.VrapObjectType
+import io.vrap.rmf.codegen.types.VrapTypeProvider
+import io.vrap.rmf.raml.model.resources.Method
+import org.eclipse.emf.ecore.EObject
+import javax.inject.Inject
+
+class JavaCommandsRenderer @Inject constructor(override val vrapTypeProvider: VrapTypeProvider) : MethodRenderer, JavaObjectTypeExtensions, EObjectExtensions {
+
+    override fun render(type: Method): TemplateFile {
+        val vrapType = vrapTypeProvider.doSwitch(type as EObject) as VrapObjectType
+        
+        val content = """
+                |package ${vrapType.`package`};
+                |
+                |import com.commercetools.importer.commands.SphereRequest;
+                |import com.fasterxml.jackson.databind.JavaType;
+                |import io.sphere.sdk.http.HttpResponse;
+                |import io.sphere.sdk.client.HttpRequestIntent;
+                |import io.sphere.sdk.http.HttpMethod;
+                |import io.sphere.sdk.json.SphereJsonUtils;
+                |import io.sphere.sdk.client.SphereRequestUtils;
+                |
+                |final class ${type.toRequestName()} implements ${type.sphereRequestImplement()} {
+                |   
+                |   <${type.fields()}>
+                |   
+                |   <${type.constructor()}>
+                |   
+                |   <${type.ofMethod()}>
+                |
+                |   @Override
+                |   public HttpRequestIntent httpRequestIntent() {
+                |      ${type.httpRequestIntentBody()}
+                |   }
+                |   
+                |   @Override
+                |   public ${type.javaReturnType()} deserialize(HttpResponse httpResponse) {
+                |       return SphereRequestUtils.deserialize(httpResponse, jacksonJavaType());
+                |   }   
+                |   
+                |   <${type.jacksonJavaTypeMethod()}>
+                |}
+                |
+                """.trimMargin().keepIndentation()
+        
+        return TemplateFile(
+            relativePath = "${vrapType.`package`}.${type.toRequestName()}".replace(".", "/") + ".java",
+            content = content
+        )
+    }
+
+    private fun Method.httpRequestIntentBody() : String? {
+        val path = this.resource().fullUri.template
+        val parts = path.split("/").filter { it.isNotEmpty() }
```

Comment:
I would suggest to use the methods that the `UriTemplate` already provides instead of creating your own simple paser here. WDYT?